sqlalchemy_redshift/dialect.py

```
@@ -981,6 +981,191 @@ def _get_table_or_view_names(self, relkind, connection, schema=None, **kw):
+            else:
+                coltype = None
+                break
+
+        if coltype:
+            coltype = coltype(*args, **kwargs)
+            if is_array:
+                coltype = self.ischema_names["_array"](coltype)
+        elif no_format_type:
+            util.warn(
+                "PostgreSQL format_type() returned NULL for column '%s'"
+                % (name,)
```

Ruff (0.14.0)

Use format specifiers instead of percent format

Replace with format specifiers

(UP031)